src/Draco.Compiler/Semantics/Symbol.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+
+namespace Draco.Compiler.Semantics;
+
+/// <summary>
+/// Defines semantic symbols.
+/// </summary>
+public static class Symbol
+{
+    /// <summary>
+    /// A base symbol.
+    /// </summary>
+    public interface ISymbol
+    {
+        /// <summary>
+        /// The name of the symbol.
+        /// </summary>
+        string Name { get; }
+    }
+
+    /// <summary>
+    /// A variable symbol.
+    /// </summary>
+    public interface IVariable : ISymbol
```

Comment:
The name of this symbol?

Reply:
Ah missed it, thanks

Reply:
That raises a question if we will have unnamed symbol, like the unreachable type or types resulting from complex type expressions like `[]int32`

Reply:
Unreachable could just have an unspeakable name (ex. `<>_unreachable`) and its own symbol (ex. `IUnreachableType`), and array types could also have their own symbol and literally just the name `[]T`. Question is what the point of the names actually is, for unreachable an unspeakable name works just fine and for for arrays they're a constructed type anyway, it's also kind of unspeakable.

Reply:
Alright, let's go with this design for now